crates/noirc_evaluator/src/ssa/acir_gen/operations/sort.rs

```diff
@@ -19,19 +19,40 @@ pub fn evaluate_permutation(
     out_expr: &Vec<Expression>,
     evaluator: &mut Evaluator,
 ) -> Vec<Witness> {
-    let (w, b) = permutation_layer(in_expr, evaluator);
-    // we constrain the network output to out_expr
+    let bits = Vec::new();
+    let (w, b) = permutation_layer(in_expr, &bits, true, evaluator);
+    // we contrain the network output to out_expr
     for (b, o) in b.iter().zip(out_expr) {
         evaluator.opcodes.push(AcirOpcode::Arithmetic(subtract(b, FieldElement::one(), o)));
     }
     w
 }
 
+// Same as evaluate_permutation() but uses the provided witness as network control bits
+pub fn evaluate_permutation_with_witness(
+    in_expr: &Vec<Expression>,
+    out_expr: &Vec<Expression>,
+    bits: &Vec<Witness>,
+    evaluator: &mut Evaluator,
+) {
+    let (w, b) = permutation_layer(in_expr, bits, false, evaluator);
+    debug_assert_eq!(w, *bits);
```

Comment:
We should change this to a regular assert_eq

Reply:
I disagree, the code ensures that this is the case so there is no point to check this every time, especially when the arrays are big.
This is more like a test, I can remove it considering that the unit test is already checking for this.